samples/python/multinomial_causal_lm/multinomial_causal_lm.py

```diff
@@ -93,11 +94,14 @@ def put(self, token_id: int) -> bool:
         elif len(text) >= 3 and text[-3:] == chr(65533):
             # Don't print incomplete text.
             pass
-        elif len(text) > self.print_len:
+        elif len(text) > self.print_len + delay_n_chars:
             # It is possible to have a shorter text after adding new token.
             # Print to output only if text length is increaesed.
-            word = text[self.print_len:]
-            self.print_len = len(text)
+            # Also, in some cases adding the next token can shorten the text, 
+            # e.g. when apostrophe removing regex had worked after adding new tokens.
```

Comment:
```suggestion
            # E.g. when apostrophe removing regex had worked after adding new tokens.
```
This statements are equivalent:
> It is possible to have a shorter text after adding new token.

and
> in some cases adding the next token can shorten the text